File: ShrimpBuddy-Android/app/src/main/java/com/shrimpbuddy/config/APIConfig.kt
package com.shrimpbuddy.config

object APIConfig {
    // ─── Set this to your web backend base URL ────────────────────────────────
    // 10.0.2.2 is the Android emulator's alias for host machine localhost
    // If running on a physical device, use your computer's local IP (e.g. http://192.168.x.x:8085/api/v1/)
    const val BASE_URL = "http://10.0.2.2:8085/api/v1/"
    // ──────────────────────────────────────────────────────────────────────────

    object Endpoints {
        // Auth
        const val LOGIN           = "auth/login"
        const val REGISTER        = "auth/register"
        const val FORGOT_PASSWORD = "auth/forgot-password"
        const val RESET_PASSWORD  = "auth/reset-password"
        // Dashboard
        const val DASHBOARD       = "dashboard"
        // Sections
        const val SECTIONS        = "sections"
        // Ponds
        const val PONDS           = "ponds"
        fun pondDetail(id: String) = "ponds/$id"
        // Feed
        const val FEED_LOGS       = "feed-logs"
        const val FEED_INVENTORY  = "feed-inventory"
        const val FEED_DISPATCH   = "feed-dispatch"
        // Chemicals
        const val CHEMICAL_USAGE  = "chemical-usage"
        const val CHEMICAL_INV    = "chemical-inventory"
        // Sampling
        const val SAMPLING        = "sampling-logs"
        // Water
        const val WATER_PARAMS    = "water-parameters"
        fun waterParams(pondId: String) = "water-parameters/$pondId"
        // Finance
        const val FINANCE         = "finance"
        // Harvest
        const val HARVEST         = "harvest-forecasts"
        // Market
        const val MARKET          = "market-prices"
        // Reports
        const val REPORTS         = "reports"
        // Staff
        const val USERS           = "users"
        const val AUDIT           = "audit-logs"
        // Settings
        const val SETTINGS        = "settings"
    }
}

File: ShrimpBuddy-iOS/ShrimpBuddy/Config/APIConfig.swift
import Foundation

enum APIConfig {
    // ─── Set this to your web backend base URL ───────────────────────────────
    static let baseURL = "http://localhost:8085/api/v1"
    // ─────────────────────────────────────────────────────────────────────────

    enum Endpoints {
        // Auth
        static let login              = "\(baseURL)/auth/login"
        static let register           = "\(baseURL)/auth/register"
        static let forgotPassword     = "\(baseURL)/auth/forgot-password"
        static let resetPassword      = "\(baseURL)/auth/reset-password"
        // Dashboard
        static let dashboard          = "\(baseURL)/dashboard"
        // Sections
        static let sections           = "\(baseURL)/sections"
        // Ponds
        static let ponds              = "\(baseURL)/ponds"
        static func pondDetail(_ id: String) -> String { "\(baseURL)/ponds/\(id)" }
        // Feed
        static let feedLogs           = "\(baseURL)/feed-logs"
        static let feedInventory      = "\(baseURL)/feed-inventory"
        static let feedDispatch       = "\(baseURL)/feed-dispatch"
        // Chemicals
        static let chemicalUsage      = "\(baseURL)/chemical-usage"
        static let chemicalInventory  = "\(baseURL)/chemical-inventory"
        // Sampling
        static let samplingLogs       = "\(baseURL)/sampling-logs"
        // Water
        static let waterParameters    = "\(baseURL)/water-parameters"
        static func waterParams(_ pondId: String) -> String { "\(baseURL)/water-parameters/\(pondId)" }
        // Finance
        static let finance            = "\(baseURL)/finance"
        // Harvest
        static let harvest            = "\(baseURL)/harvest-forecasts"
        // Market
        static let marketPrices       = "\(baseURL)/market-prices"
        // Reports
        static let reports            = "\(baseURL)/reports"
        // Staff / User Management
        static let users              = "\(baseURL)/users"
        static func userDetail(_ id: String) -> String { "\(baseURL)/users/\(id)" }
        static let auditLogs          = "\(baseURL)/audit-logs"
        // Settings
        static let settings           = "\(baseURL)/settings"
        // Price List (feed/chemical master records)
        static let feedMaster         = "\(baseURL)/feed-master"
        static let chemicalMaster     = "\(baseURL)/chemical-master"
        // Dispatches (used for Finance cost calculation)
        static let feedDispatches     = "\(baseURL)/feed-dispatch"
        static let chemicalDispatches = "\(baseURL)/chemical-dispatches"
        // Other farm expenses
        static let farmExpenses       = "\(baseURL)/farm-expenses"
        static func farmExpenseDetail(_ id: String) -> String { "\(baseURL)/farm-expenses/\(id)" }
        // Audit & Compliance
        static let auditPolicies      = "\(baseURL)/audit-policies"
        static let complianceIssues   = "\(baseURL)/compliance-issues"
    }
}
